atlasdb-commons/src/main/java/com/palantir/common/concurrent/ConcurrentStreams.java

```diff
@@ -56,36 +54,38 @@ private ConcurrentStreams() {}
             return values.stream().map(mapper);
         }
 
-        Map<T, CompletableFuture<S>> futuresByValue = values.stream()
-                .collect(Collectors.toMap(Function.identity(), value -> new CompletableFuture<>()));
-        Queue<T> valueQueue = new ConcurrentLinkedQueue<T>(values);
+        List<StreamElement<T, S>> elements = values.stream()
+                .map((Function<T, StreamElement<T, S>>) StreamElement::new)
```

Comment:
maybe nit: Intellij complains here due to confusion between generic types and Object. 

After the map we have a `Stream<StreamElement<T, Object>>`, but we are casting the Object to S.

Reply:
Yeah I originally did this since Intellij suggested it but agreed the casting is probably silly. Will switch to something simpler like `value -> new StreamElement<T,S>(value)` to make it clear.